### DISCUSSION TITLE
Flask blueprint without a static_folder

### DISCUSSION THREAD

**Discussion by hollabigj** [pallets/flask#4608] [Q&A]
Hi,

What if the business/corporation/or senior architecture/designer do not want to have a static folder in the directory. IE static_folder

Would it possible to add static_css and static_javascript or override the constructor to point each desirable directory paths without strictly adhere static_folder or have a static folder in the parent directory.

So, it can serve css and javascript without a static folder.

For example:

index = Blueprint('index', __name__, template_folder=html_dir, static_css=css_dir, static_js=js_dir)

This cannot be achieved:

index = Blueprint('index', __name__, template_folder=html_dir, static_folder=multi_dir)

multi_dir points to the parent directory and html css link and script points to css and js/lib directories. It does not work at all.

If it can be overrided, how?

**Comment by tachyondecay** [comment]
Nothing stops you from doing this already—you just wouldn’t be able to generate URLs dynamically and would instead have to hardcode them.

If you want to use `url_for`, then you will need to add new endpoints that correspond to your additional folders.

If you want to do this at the blueprint level, you can sublcass the `Blueprint` class and add your custom URL rules there. See the following lines from `Blueprint.register()`, which is where blueprints register their static folder endpoints (but you could probably get away with just overriding `Blueprint.__init__()`

https://github.com/pallets/flask/blob/1888df34405d46ac0e7524d0de9408e37449feaa/src/flask/blueprints.py#L325-L330
